mvp-spec/prebid-implementation.md

```diff
@@ -0,0 +1,691 @@
+```
+PAF lib will ingest the response and construct the full audit log by appending the PAF data and
+seed (which if can look up via pafTransactionId) to the response, resulting in:
+
+<!--partial-begin { "files": [ "audit-log.json" ], "block": "json" } -->
+<!-- ⚠️ GENERATED CONTENT - DO NOT MODIFY DIRECTLY ⚠️ -->
+```json
+{
+    "data": {
+        "identifiers": [
+            {
+                "version": 0,
+                "type": "prebid_id",
+                "value": "7435313e-caee-4889-8ad7-0acd0114ae3c",
+                "source": {
+                    "domain": "operotor0.com",
+                    "timestamp": 1639589531,
+                    "signature": "12345_signature"
+                }
+            }
+        ],
+        "preferences": {
+            "version": 0,
+            "data": { 
+                "opt_in": true 
+            },
+            "source": {
+                "domain": "cmp1.com",
+                "timestamp": 1639589531,
+                "signature": "12345_signature"
+            }
+        }
+    },
+    "seed": {
+        "version": 0,
+        "transaction_id": "a0651946-0f5b-482b-8cfc-eab3644d2743",
+        "publisher": "publisher.com",
+        "source": {
+          "domain": "ad-server.com",
+          "timestamp": 1639589531,
+          "signature": "12345_signature"
+        }
+    },
+    "transmissions": [
+        {
+            "version": 0,
+            "receiver": "ssp1.com",
+            "status": "success",
+            "details": "",
+            "source": {
+                "domain": "ssp1.com",
+                "timestamp": 1639589531,
+                "signature": "12345_signature"
+            }
+        },
+        {
+            "version": 0,
+            "receiver": "ssp2.com",
+            "status": "success",
+            "details": "",
+            "source": {
+                "domain": "ssp2.com",
+                "timestamp": 1639589531,
+                "signature": "12345_signature"
+            }
+        },
+        {
+            "version": 0,
+            "receiver": "dsp.com",
+            "status": "success",
+            "details": "",
+            "source": {
+                "domain": "dps.com",
+                "timestamp": 1639589531,
+                "signature": "12345_signature"
+            }
+        }
+    ]
+}
+```
+<!--partial-end-->
+
+PAF lib will store the audit log keyed by the `prebidTransacionId`.  PAF Lib will also store a mapping
+of `slotId` to the `prebidTransacionId`. If there is an existing slot mapping it will be overridden.
+
+PAF will expose the audit log to the window with two methods of retrieval.
+```
+// look up by transactionId
+PAF.getAuditLogByTransaction(transactionId)
+
+// look up transactionId by slot mapping
+PAF.getAuditLogBySlot(slotId)
+```
+
+This iteration is only focused on audit logs for winner. This results in a simplified trigger.
+
+```
+window.pbjs.onEvent("bidWon", (bid) =>
```

Comment:
I'm not sure exactly where this code will be executed: above you say
> PAF will register a trigger to retrieve transmission responses explained below.
The trigger will perform a callback to PAF lib to register a transmission
response.

It sounds like PAF itself would register the trigger, but that surprises me. I would expect PAF lib to provide the _trigger_ as you call it, but not to register it to PBJS.
I guess the publisher (or the PBJS adapter, sorry for not being very comfortable with this part 😉 ) would do the "registering". Am I correct? If so, can you please slightly rephrase?

Reply:
After offline discussion: yes, PAF will register the trigger.
If `pbjs` doesn't exist in the window scope, then it will ignore it.